packages/eslint-plugin/src/rules/member-ordering.ts

```diff
@@ -615,6 +615,39 @@ function getRank(
   return getRankOrder(memberGroups, orderConfig);
 }
 
+function changeToGroups(
+  memberSet: Member[],
+  memberType: MemberType[],
+  supportsModifiers: boolean,
+): Member[][] {
+  const groupedMembers: Member[][] = [];
+
+  const getPreviousRank = (): number | undefined => {
+    if (groupedMembers.at(-1)?.at(0) === undefined) {
+      return undefined;
+    }
+    return getRank(
+      groupedMembers[groupedMembers.length - 1][0],
+      memberType,
+      supportsModifiers,
+    );
+  };
+
+  memberSet.forEach((member, index) => {
+    if (getRank(member, memberType, supportsModifiers) === getPreviousRank()) {
+      groupedMembers.at(-1)?.push(member);
+    } else if (index === memberSet.length - 1) {
+      return;
+    } else if (
+      getRank(member, memberType, supportsModifiers) ===
+      getRank(memberSet.at(index + 1)!, memberType, supportsModifiers)
```

Comment:
[Style] No need to `.at`. The types for it are always nullable, but since we know it's defined, we can just `[index + 1]`.

```suggestion
      getRank(memberSet[index + 1], memberType, supportsModifiers)
```